packages/roosterjs-content-model-plugins/lib/autoFormat/link/createLinkAfterSpace.ts

```diff
@@ -15,22 +15,30 @@ export function createLinkAfterSpace(editor: IEditor) {
             const length = selectedSegmentsAndParagraphs[0][1].segments.length;
             const marker = selectedSegmentsAndParagraphs[0][1].segments[length - 1];
             const textSegment = selectedSegmentsAndParagraphs[0][1].segments[length - 2];
+
             if (
                 marker.segmentType == 'SelectionMarker' &&
                 textSegment.segmentType == 'Text' &&
                 !textSegment.link
             ) {
                 const link = textSegment.text.split(' ').pop();
-                if (link && matchLink(link)) {
+                const url = link?.trimRight();
+                if (url && link && matchLink(url)) {
                     textSegment.text = textSegment.text.replace(link, '');
```

Comment:
In a corner case, for example, the text contains two same url strings:
www.bing.com www.bing.com

This will only replace the first one. Actually we need to replace the last one.